### ISSUE TITLE
Test case 1.2 failed

### ISSUE THREAD

**Issue by Wenxing-hou** [DMTF/SPDM-Responder-Validator#43]
When use the latest libspdm, the test case 1.2 failed.

```
test suite (spdm_responder_conformance_test) - pass: 2563, fail: 2
test group 1 (spdm_test_group_version) - pass: 11, fail: 1
  test case 1.1 (spdm_test_case_version_success) - pass: 7, fail: 0
  test case 1.2 (spdm_test_case_version_invalid_request) - pass: 4, fail: 1
```

**Comment by steven-bellock** [comment]
libspdm now returns `SPDM_ERROR_CODE_VERSION_MISMATCH` instead of `SPDM_ERROR_CODE_INVALID_REQUEST`. But regardless of the error code the 1.2 test shouldn't exist, as the SPDM specification doesn't say what should be done in this case.

https://github.com/DMTF/libspdm/blob/4f8f99005368cb0e0091ef07b693631593af9b97/library/spdm_responder_lib/libspdm_rsp_version.c#L38-L42

**Comment by Wenxing-hou** [comment]
> libspdm now returns `SPDM_ERROR_CODE_VERSION_MISMATCH` instead of `SPDM_ERROR_CODE_INVALID_REQUEST`. But regardless of the error code the 1.2 test shouldn't exist, as the SPDM specification doesn't say what should be done in this case.
> 
> https://github.com/DMTF/libspdm/blob/4f8f99005368cb0e0091ef07b693631593af9b97/library/spdm_responder_lib/libspdm_rsp_version.c#L38-L42

Yes. The spec doesn't  clear this behavior. So I think we can change return value for this issue and keep the check until there are more detail behavior in spec.

**Comment by jyao1** [comment]
I think we can keep both values be valid.

**Comment by jyao1** [comment]
please update document as well.
